src/agent/Server.cpp

```diff
@@ -0,0 +1,62 @@
+#include "Server.h"
+#include <qnetworkinterface.h>
+#include <qdatastream.h>
+#include <qtcpsocket.h>
+#include <iostream>
+#include "common/constants.hpp"
+
+
+Server::Server(QObject * parent)
+{
+    connect(&tcpServer, &QTcpServer::newConnection, this, &Server::SendData);
+}
+
+bool Server::Init()
+{
+        if (!tcpServer.listen(QHostAddress::Any, RDHMPort)) {
+            return false;
+        }
+
+        QString ipAddress;
```

Comment:
QTcpServer::serverAddress()  ?

Reply:
Nope. This method returns 0.0.0.0 always